src/mpv.rs

```diff
@@ -158,6 +158,7 @@ impl<'parent> Iterator for MpvNodeMapIter<'parent> {
     }
 }
 
+#[derive(Clone, Debug)]
```

Comment:
I don't think we can add `Clone` at this point and at the `Iterator`s, because only the pointers would be cloned, not the actual content.
Then the `Drop` implementation may get called multiple times, or one Iterator may try to access an already freed `MpvNode`.

One option is to manually implement a deep `Clone`.

Reply:
Good catch! I completely missed this. Your deep copy suggestion is not obvious to me, but I shall think about it some more.

Reply:
Here are the approaches I was considering:

1. `MpvNode(Rc<InnerMpvNode(mpv_node)>)` - Basically, wrap in an `Rc`. The `InnerMpvNode` implements `Drop`, `Rc` lets us `clone()` and delays drop until the last copy is dropped, and `MpvNode` actually implements the API. This avoids a deep copy. It's a lot of wrapping for this, but I think it would work.

2. Use `libmpv` to copy the data - Unfortunately, I don't think they expose a public method to copy the data, so I don't think this is possible. 

3. Deep copy the data into, perhaps, a new `MpvNode` allocated by Rust. This would require traversing the `MpvNode` tree data structure and copying all the data. The resulting `MpvNode` could not be `mpv_free()`ed, so we'd somehow have to do `MpvNode::Borrowed` versus `MpvNode::Owned`, where only`MpvNode::Borrowed` calls `mpv_free()` on `Drop`. 

After thinking about it, I am not sure it should impl `Clone`. This is event data that's only valid for as long as the `fn` handling it. Would people want to `clone()` it opaquely without matching against the inner values? If matching is painful, maybe we should put our effort into improving that.

Reply:
Yeah, I agree it is probably not worth it. With the breaking change required (taking `&mut self` in `wait_event`) it is not a big issue that this is breaking, we can combine them in a single release.